IF_lesson.py

```diff
@@ -3,7 +3,7 @@
 
 def string_comprasion(string_1, string_2):
 
-    if type(string_1) and type(string_2) is str:
```

Comment:
кстати, как ты думаешь, а почему ```type(string_1) and type(string_2) is str``` может некорректно работать?

Reply:
Сложно... явно знаний недостаточно. Из гугла понял то что isinstance корректнее обрабатывает классы которые наследуются и вообще создана для проверки, а type...
***
Хотя функция type() возвращает тип переданного аргумента, с ее помощью можно проверить, принадлежит ли аргумент тому или иному типу:
***
Но придумать в каком случае... если тип будет str чтобы type её не верно обработала не могу.

Reply:
Тут можно использовать тайп, но тогда надо по-другому выражение под мыли писать, вопрос как?
Подсказка: and оператор над булевыми значениями

Reply:
```suggestion
    if type(string_1) and type(string_2) is str:
```
хм... я тут забыл протестить вариант с первым интом а вторым стрингом... поэтому проверяется только второе значение, а первое возвращая хоть что то кроме False складывается с ним... нужно тогда так:
```
 if type(string_1) is str and type(string_2) is str:
```

Reply:
ага, именно
только не только False, но и пустая строка или пустой лист не пройдет проверку